GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs

```diff
@@ -45,19 +47,125 @@ public CaseContentBuilder(AllureClientWrapper allureClientWrapper, Context conte
+    {
+        var extractAttachments = ExtractAttachments(scenario.Steps.ToList());
+
+        var background = featureFile.Document.Feature.Children.OfType<Background>().FirstOrDefault();
+        if (background is not null)
+        {
+            var backgroundAttachments = ExtractAttachments(background.Steps.ToList());
+            return backgroundAttachments.Concat(extractAttachments).ToDictionary(pair => pair.Key, pair => pair.Value);
+        }
+
+        return extractAttachments;
+    }
+
+    private Dictionary<int, ByteArrayPart> ExtractAttachments(List<Step> steps)
+    {
+        var result = new Dictionary<int, ByteArrayPart>();
+
+        for (var index = 0; index < steps.Count; index++)
+        {
+            var step = steps[index];
+
+            switch (step.Argument)
+            {
+                case DocString docString:
+                    
+                    var textBytes = Encoding.ASCII.GetBytes(docString.Content);
+                    result.Add(index, new ByteArrayPart(textBytes, "Text", "text/plain"));
+                    break;
+                
+                case DataTable table:
+                    
+                    var csvBytes = Encoding.ASCII.GetBytes(ConvertToCsvTable(table.Rows.ToList()));
+                    result.Add(index, new ByteArrayPart(csvBytes, "Table", "text/csv"));
+                    break;
+            }
+        }
+
+        return result;
+    }
+
+    private Quantori.AllureTestOpsClient.Model.Scenario AddScenario(Scenario scenario, IFeatureFile featureFile)
+    {
+        var steps = GetSteps(scenario, featureFile);
+        if (!steps.Any()) return null;
+
+        var allureScenario = new Quantori.AllureTestOpsClient.Model.Scenario
+        {
+            Steps = new List<Quantori.AllureTestOpsClient.Model.Step>()
         };
+        allureScenario.Steps.AddRange(steps);
+        return allureScenario;
+    }
+
+    private List<Quantori.AllureTestOpsClient.Model.Step> GetSteps(Scenario scenario, IFeatureFile featureFile)
+    {
+        var scenarioSteps = ExtractSteps(scenario.Steps.ToList());
 
-        return caseRequest;
+        var background = featureFile.Document.Feature.Children.OfType<Background>().FirstOrDefault();
+        if (background is not null)
+        {
+            var backgroundSteps = ExtractSteps(background.Steps.ToList());
+            return backgroundSteps.Concat(scenarioSteps).ToList();
+        }
+
+        return scenarioSteps;
+    }
+
+    private List<Quantori.AllureTestOpsClient.Model.Step> ExtractSteps(List<Step> steps)
+    {
+        var allureSteps = new List<Quantori.AllureTestOpsClient.Model.Step>();
+        foreach (var step in steps)
+        {
+            var allureStep = new Quantori.AllureTestOpsClient.Model.Step
+            {
+                Keyword = step.Keyword.Trim(),
+                Name = step.Text
+            };
+
+            allureSteps.Add(allureStep);
+        }
+
+        return allureSteps;
```

Comment:
Here you can easily use Linq Select method and return result in one command
And to make it more convenient you can add constructor to `Quantori.AllureTestOpsClient.Model.Step` for keyword and name

Reply:
Done